include/sway/config.h

```diff
@@ -110,12 +110,19 @@ struct bar_config {
 	 * Always visible in dock mode. Visible only when modifier key is held in hide mode.
 	 * Never visible in invisible mode.
 	 */
-	char *mode;
+	char *display_mode;
```

Comment:
Would it make more sense to stick with `mode` as the varname for `mode` config keyword? that seems like how the rest of the vars work. 

binding mode is not really a configuration option for bar and there's no config word for it, it's just a label published by sway itself for display. Just incidental that its stored in bar_config, because it doubles as a state object as well as a config store.

Would also make the patch smaller.